src/main/java/org/apache/mesos/scheduler/plan/ReconciliationStrategy.java

```diff
@@ -23,9 +24,14 @@ private ReconciliationStrategy(final ReconciliationPhase phase) {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(Collection<String> dirtiedAssets) {
         // ReconciliationPhases only have a single Block
-        return Optional.of(phase.getBlock(0));
+        for (Block block : phase.getBlocks()) {
```

Comment:
`null` check `dirtiedAssets` ? Also, if `dirtiedAssets` is empty or null, just `return Optional.of(phase.getBlock(0));` ?